src/base/InstanceId.h

```diff
@@ -39,15 +38,16 @@ class InstanceId
     const char * prefix() const;
 
 public:
-    Value value = 0; ///< instance identifier
+    Value value = Value(); ///< instance identifier
 
 private:
     InstanceId(const InstanceId &); ///< not implemented; IDs are unique
     InstanceId& operator=(const InstanceId &); ///< not implemented
 };
 
-/// convenience macro to instantiate Class-specific stuff in .cc files
-#define InstanceIdDefinitions(Class, pfx) \
+/// 1-parameter instantiation macro (for private use only).
+/// In other sources use InstanceIdDefinitions() instead.
+#define InstanceIdDefinitions2(Class, pfx) \
```

Comment:
Can we change macro definition order and then do something like this instead:
```suggestion
#define InstanceIdDefinitions2(Class, pfx) InstanceIdDefinitions3(Class, int, pfx)
```

Reply:
Done.

Reply:
Done.